service/src/test/java/bio/terra/cbas/runsets/inputs/TestInputGeneratorBuildInputs.java

```diff
@@ -169,4 +173,56 @@ private static RecordResponse fooRatingRecord(String rawAttributeJson)
             .trim(),
         RecordResponse.class);
   }
+
+  @Test
+  void invalidAttribute() throws Exception {
+    String incorrectAttributeDefinition =
+        """
+            [
+              {
+                "input_name": "lookup_foo",
+                "input_type": { "type": "primitive", "primitive_type": "String" },
+                "source": {
+                  "type": "record_lookup",
+                  "record_attribute": "foo_rating"
+                }
+              }
+            ]
+          """
+            .stripIndent()
+            .trim();
+
+    RecordResponse recordResponse =
+        objectMapper.readValue(
+            """
+        {
+          "id": "FOO1",
+          "type": "FOO",
+          "attributes": {
+            "MY_RECORD_ATTRIBUTE": "Hello, world"
+          },
+          "metadata": {
+            "provenance": "TODO: RECORDMETADATA"
+          }
+        }
+        """
+                .trim()
+                .stripIndent(),
+            RecordResponse.class);
+
+    List<WorkflowInputDefinition> inputDefinitions =
+        objectMapper.readValue(incorrectAttributeDefinition, new TypeReference<>() {});
+
+    Exception exception = null;
+
+    try {
+      InputGenerator.buildInputs(inputDefinitions, recordResponse);
+    } catch (Exception e) {
+      exception = e;
+    }
+
+    assertNotNull(exception);
+    assertTrue(exception instanceof WorkflowAttributesNotFoundException);
+    assertEquals(exception.getMessage(), "Attribute not found in WDS record.");
```

Comment:
This might look nicer as an `assertThrows` (something like in [this answer](https://stackoverflow.com/questions/40268446/junit-5-how-to-assert-an-exception-is-thrown))